xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingExchange.java

```diff
@@ -18,47 +17,72 @@
 public class KrakenStreamingExchange extends KrakenExchange implements StreamingExchange {
 
     private static final Logger LOG = LoggerFactory.getLogger(KrakenStreamingExchange.class);
-
     private static final String API_URI = "wss://ws.kraken.com";
+    private static final String API_AUTH_URI = "wss://ws-auth.kraken.com";
+    private static final String API_BETA_URI = "wss://beta-ws.kraken.com";
 
-    private final KrakenStreamingService streamingService;
+    private KrakenStreamingService streamingService, privateStreamingService;
     private KrakenStreamingMarketDataService streamingMarketDataService;
-    
+    private KrakenStreamingTradeService streamingTradeService;
+
     private final SynchronizedValueFactory<Long> nonceFactory = new CurrentTimeNonceFactory();
 
     public KrakenStreamingExchange() {
-        this.streamingService = new KrakenStreamingService(API_URI);
+    }
+
+    private static String pickUri(boolean isPrivate, Boolean useSandbox) {
+        if (useSandbox != null && useSandbox)
```

Comment:
Is the beta environment really a "sandbox"?

Reply:
Not clear: but I changed the flag to Use_Beta
https://support.kraken.com/hc/en-us/articles/360000919926-Does-Kraken-offer-a-Test-API-or-Sandbox-Mode-